app/Modules/Availability/AvailabilityService.php

```diff
@@ -0,0 +1,159 @@
+     * @return bool
+     */
+    public function validateAvailabilityTimes(Carbon $from, Carbon $to, User $user, int $doNotCheck =  NULL): bool
+    {
+        if ($from->isAfter($to)) {
+            return false;
+        }
+
+        if ($from->isBefore(new Carbon())) {
+            throw new AvailabilityInPastException();
+        }
+
+        if ($from->diffInMinutes($to) < self::MIN_BLOCK_TIME) {
+            throw new AvailabilityMinimumTimeException(self::MIN_BLOCK_TIME);
+        }
+
+        $avail = Availability::where([['user_id', $user->id], ['id', '!=', $doNotCheck]]);
+
+        // Find between the times being booked for
+        $avail->where(function ($query) use (&$from, &$to) {
+            // Where the start date is inside the booked time
+            $query->where(function ($query) use (&$from, &$to) {
+                $query->where('from', '>', $from)
+                    ->where('from', '<', $to);
+                // Or where the end date is inside the booked time
+            })->orWhere(function ($query) use (&$from, &$to) {
+                $query->where('to', '>', $from)
+                    ->where('to', '<', $to);
+                // Or where the times are the same
+            })->orWhere(['from' => $from, 'to' => $to]);
+        });
+
+        return $avail->doesntExist();
+    }
+
+    public function createAvailability(array $availabilityData): Availability
+    {
+        ['from' => $from, 'to' => $to, 'user_id' => $userId] = $availabilityData;
+
+        $availabilityUser = $this->user::findOrFail($userId);
+
+        try {
+            $passesTimeChecks = $this->validateAvailabilityTimes($from, $to, $availabilityUser);
+        } catch (Exception $e)  {
+            if (get_class($e) == 'AvailabilityInPastException') {
+                throw new AvailabilityInPastException();
+            } else {
+                throw new AvailabilityMinimumTimeException(self::MIN_BLOCK_TIME);
+            }
+        }
+
+        if (!$passesTimeChecks) {
+            throw new OverlappingAvailabilityException();
+
+        }
+        return $availabilityUser->availability()->create([
+            'user_id' => $availabilityUser->id,
+            'from' => $from,
+            'to' => $to,
+        ]);
+    }
+
+    public function updateExistingAvailability(array $newData, bool $adminOverride = false): bool
+    {
+        ['from' => $from, 'to' => $to, 'user_id' => $userId] = $newData;
+
+        /** @var Availability $existingAvailability */
+        $existingAvailability = Availability::findOrFail($newData['id']);
+
+        $availabilityUser = $this->user::findOrFail($userId);
+
+        if (!$adminOverride && ($userId != $existingAvailability['user_id'])) {
+            throw new PrivilegeException('update this availability');
+        }
+
+        try {
+            $passesTimeChecks = $this->validateAvailabilityTimes($from, $to, $availabilityUser, $existingAvailability->getKey());
+        } catch (Exception $e)  {
```

Comment:
Same here, the potential for multiple catch blocks.

Reply:
done